Add "Create rollup job" wizard foundation and logistics step

cjcenizal opened the pull request:
To-do:

- [x] Wizard framework
- [x] Logistics step with validation
- [x] Display a spinner to block user interaction when saving a new job
- [x] Show inline feedback if job already exists
- [x] Remove dependency upon `redux-actions`

To-do in later PRs:

- [ ] Subsequent steps
- [x] Show matching indices when user enters index pattern, or show suggestions
- [ ] User-friendly cron editor

## Default

![image](https://user-images.githubusercontent.com/1238659/44610513-501e1980-a7b1-11e8-9dbf-63d72f200afd.png)

## Errors

### Step validation

![image](https://user-images.githubusercontent.com/1238659/44667317-4e31a180-a9cf-11e8-8d6b-bfe8a3c77ba6.png)

### Inline feedback for recognized API errors

![image](https://user-images.githubusercontent.com/1238659/44667327-5689dc80-a9cf-11e8-9dca-7dcef099a72a.png)

### Toasts for unrecognized API errors

![image](https://user-images.githubusercontent.com/1238659/44667363-702b2400-a9cf-11e8-8634-22e7a7a3e4e4.png)

![image](https://user-images.githubusercontent.com/1238659/44667366-73beab00-a9cf-11e8-9510-4b994162467f.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/900/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/904/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/914/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/923/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/930/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/933/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/981/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1382/)

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/sections/job_create/step_logistics/step_logistics.js:
```diff
@@ -0,0 +1,285 @@
+            label={(
+              <FormattedMessage
+                id="xpack.rollupJobs.create.stepLogistical.fieldName.label"
+                defaultMessage="Rollup job name"
```
Good catch. I'll update the text to refer to it as an "ID" since this is how the docs and API refer to it, check the acceptable values with Zach, and implement client-side validation.

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/sections/job_create/step_logistics/step_logistics.js:
```diff
@@ -0,0 +1,285 @@
+      rollupPageSize: errorRollupPageSize,
+    } = fieldErrors;
+
+    const illegalCharacters = ['\\', '/', '?', '"', '<', '>', '|'].join(', ');
```
Good idea.

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/sections/job_create/job_create.js:
```diff
@@ -0,0 +1,444 @@
+        );
+      }
+
+      if (!indexPattern || !indexPattern.trim()) {
```
Yes, good catch.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1396/)

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/services/query_params.js:
```diff
@@ -0,0 +1,20 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function extractQueryParams(queryString) {
```
Correct, react-router only supplies the search string, without parsing it into the individual query keys and values (https://github.com/ReactTraining/react-router/issues/4410). We could use a lib instead of hand-rolling this, but I think I want to avoid introducing yet another dependency.

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/store/store.js:
```diff
@@ -18,4 +18,6 @@ export const rollupJobsStore = (initialState = {}) => {
-};
+}
+
+export const rollupJobsStore = createRollupJobsStore();
```
Good catch. I looked into this and it doesn't look like the way to store is being created is the cause. I think I just need to clear this error state when this component is unmounted.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1397/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1398/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1460/)

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/sections/job_create/step_logistics/step_logistics.js:
```diff
@@ -0,0 +1,285 @@
+    return (
+      <Fragment>
+        <EuiSpacer size="m" />
+        <EuiCallOut
```
I borrowed this pattern from the wizards I worked on in Cloud. I'm definitely open to changing it though. Let's see what the design team suggests when we bring them in.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1462/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1472/)

cjcenizal commented:
@jen-huang I've addressed your feedback and also implemented a two-column layout. Could you take another look please?

![image](https://user-images.githubusercontent.com/1238659/45052325-53c86080-b03b-11e8-8e5e-5120d23371ee.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1663/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1679/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1689/)

cjcenizal commented:
Last change! Using `EuiDescribedFormGroup` for the layout.

![image](https://user-images.githubusercontent.com/1238659/45063685-03192d80-b064-11e8-80c4-520bf8c97038.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1703/)

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/store/actions/create_job.js:
```diff
@@ -0,0 +1,80 @@
+        break;
+
+      default:
+        toastNotifications.addDanger({
```
Great idea! Done.

![image](https://user-images.githubusercontent.com/1238659/45104283-9dba5080-b0e6-11e8-967b-0c328ff03d4e.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1762/)

cjcenizal commented on x-pack/plugins/rollup/public/crud_app/sections/job_create/steps_config/validate_rollup_index.js:
```diff
@@ -0,0 +1,59 @@
+      />
+    )];
+  }
+
```
Good catch, done!

![image](https://user-images.githubusercontent.com/1238659/45107622-00afe580-b0ef-11e8-8803-747ef8a4585e.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1772/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1775/)
